### PR TITLE
[CI Change] - re-enable reviewer assignment + auto test runs

### PR DESCRIPTION
**Pull request by shuyama1**
If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
```

### CODE REVIEW

**Comment by shuyama1** [comment]
Disabled new triggers, check if the PR will run the previous CI setup
/gcbrun

File: .ci/containers/membership-checker/reviewer_assignment.go
Comment (melinath) [inline .ci/containers/membership-checker/reviewer_assignment.go:40]: This section could be simplified by doing something like

```go
foundTeamReviewer := false
if hasTeamReviewer || previousAssignedReviewers != nil {
    if hasTeamReviewer {
         foundTeamReviewer = true
    }
    // check previous reviewers for team reviewers & re-request review from them & set foundTeamReviewer appropriately
}
if !foundTeamReviewer {
    // choose random reviewer
}
```

`hasTeamReviewer` would fill a role similar to getPullRequestRequestedReviewer with one critical difference: it needs to check whether any of the people who have left reviews are team reviewers - that is, if an off-team person left a review, that can't fulfill the requirement of "at least one reviewer from the team".

File: .ci/containers/membership-checker/reviewer_assignment.go
Comment (melinath) [reply .ci/containers/membership-checker/reviewer_assignment.go:24]: hmm, the [previous logic](https://github.com/GoogleCloudPlatform/magic-modules/blob/eb6d15f1358aa286b99634a38227d4e04da701e3/.ci/containers/contributor-checker/check-contributor.sh#L24) seems to have been:

- ASSIGNEE is first requested reviewer or first reviewer
- if there is an assignee (that is, at least one requested reviewer or reviewer), re-request reviews from all team reviewers and stop
- Otherwise, randomly assign someone

That makes sense - so, we do need something like requestedReviewer. I'll make a suggestion below, since I still think removing this section will be useful.

**Review by melinath — CHANGES_REQUESTED** [review]
Overall makes sense to me and I think meets our requirements. Could you add some unit tests for this?

File: .ci/containers/membership-checker/main.go
Comment (shuyama1) [reply .ci/containers/membership-checker/main.go:32]: Sorry, that's a copy typo. Will change them! good catch!

File: .ci/containers/membership-checker/membership.go
Comment (shuyama1) [reply .ci/containers/membership-checker/membership.go:11]: Make sense!

File: .ci/containers/membership-checker/reviewer_assignment.go
Comment (shuyama1) [reply .ci/containers/membership-checker/reviewer_assignment.go:168]: That's better! Will do!

File: .ci/containers/membership-checker/membership.go
Comment (shuyama1) [reply .ci/containers/membership-checker/membership.go:11]: Actually, I'm thinking that maybe we still want reviewers on vacationList to receive re-requests. This way, if any commit's been pushed to the PR, they'd still be notified and the PR should be on their review list on GH when they come back. What's your thought on this?

**Comment by shuyama1** [comment]
@melinath The PR is still WIP, and I'll let you know when it's ready to review

**Comment by shuyama1** [comment]
@melinath The PR is ready to review. I made some modification according to your comments, and added unit tests and also a GCB trigger `membership-checker-test` that ensures unit tests only runs when anything related to this feature has changed.

File: .ci/unit-tests/gcb-membership-checker-test.yml
Comment (melinath) [reply .ci/unit-tests/gcb-membership-checker-test.yml:2]: Not a blocker / could be done in a separate PR or as part of a later refactor.

**Review by melinath — APPROVED** [review]
I didn't do a manual check that it works but it all LGTM! Much more straightforward. 💯 We might want to add more tests in the future - but we can do that as issues arise.

File: .ci/unit-tests/gcb-membership-checker-test.yml
Comment (shuyama1) [reply .ci/unit-tests/gcb-membership-checker-test.yml:2]: I added it as a cloud build trigger since we need file filter and security guard (/gcbrun needed) for the tests to be triggered. I'll have to check if github action have the similar features and we can move it to github action in a separate PR later.

**Comment by shuyama1** [comment]
I'll do a clean up and push the final version later today, and run a final test tomorrow. If everything goes well, will roll out this change early next Wednesday.

**Review by melinath — APPROVED** [review]
So basically we will now auto-run the community-checker for trusted users, but it won't actually do anything other than get a green check (because they're trusted users) - makes sense!

**Comment by shuyama1** [comment]
> So basically we will now auto-run the community-checker for trusted users, but it won't actually do anything other than get a green check (because they're trusted users) - makes sense!

Yes, the major change is in [this commit](https://github.com/GoogleCloudPlatform/magic-modules/pull/7446/commits/f927cc8f1769cd12316e2e8d38d6ec37edebde70). We change from using comment control to approval based build execution for community-checker. Trusted users will be auto approved and external users will need a reviewer to approve the run on GCB. Also, a `awaiting-approval` label will be added to let reviewers know that actions are needed for the PR and it will be removed after community-contributor is approved and executed.

File: .ci/containers/membership-checker/main.go
Comment (rileykarson) [inline .ci/containers/membership-checker/main.go:57]: I find the modal binary kind of hard to follow. As a followup we may want to consider splitting into different images or binaries. I think we can override the entrypoint.

File: .ci/containers/membership-checker/Dockerfile
Comment (shuyama1) [reply .ci/containers/membership-checker/Dockerfile]: Done!

File: .ci/containers/membership-checker/reviewer_assignment.go
Comment (shuyama1) [reply .ci/containers/membership-checker/reviewer_assignment.go:91]: yep, will replace the current `contributor-checker`

File: .ci/containers/membership-checker/community.go
Comment (shuyama1) [reply .ci/containers/membership-checker/community.go:43]: Looks like they are ordered -- `Builds will be sorted by createTime, descending` based on [API doc](https://cloud.google.com/build/docs/api/reference/rest/v1/ListBuildsResponse)

**Pull request merged**
